### PR TITLE
Preparate data

### PR DESCRIPTION
**Pull request by barti2115**
Add all steps and code responsible for preparing data to EDA step.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,57 @@
...
+### Installation
+1. Clone the repository:
+    ```bash
+    git clone <repository_url>
+    cd <repository_name>
+    ```
+
+2. Create a virtual environment (recommended):
+    ```bash
+    python -m venv venv
```
Comment (jstaszel-agh) [inline README.md:17]: My venv is called venv and it is in a folder venv (which is not included in the .gitignore)

File: README.md [hunk 2]
```diff
@@ -1 +1,57 @@
...
+    source venv/bin/activate  # Activate virtual environment (Linux/macOS)
+    # OR
+    .\venv\Scripts\activate   # Activate virtual environment (Windows)
+    ```
+
+3. Install required packages using pip:
+    ```bash
+    pip install -r requirements.txt
```
Comment (jstaszel-agh) [inline README.md:25]: This approach does not ensure full reproducibility (I can install any version of the libraries I want)

File: prepare_data.sh
Comment (jstaszel-agh) [inline prepare_data.sh:1]: I didn't expect that someone is using Linux here, but that is fine, I can just run every Python file separately.

You could also use a Python file, like a run.py or main.py which has all steps one by one, we are always using the same Python venv, right?

File: download.py
Comment (jstaszel-agh) [inline download.py:1]: I got an error: zipfile.BadZipFile: Bad CRC-32 for file 'o_d_04_2023.csv'

File: read_data.py
Comment (jstaszel-agh) [inline read_data.py:1]: An error: An error occurred while reading file 'data/precipitation\2023\o_d_12_2023.csv' with encoding 'windows-1250': Error tokenizing data. C error: EOF inside string starting at row 13419

File: filter_and_map_stations.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    output_parquet_path = "./data/localization/unfiltered_localization_data.parquet"
+    filtered_output_parquet_path = "./data/localization/filtered_localization_data.parquet"
+    filtered_df_parquet_path = "./data/parquet_files/filtered_data/data.parquet"
+    
+    print(f"Processing Parquet file: {parquet_file_path}")
+    unique_stations = get_unique_stations(parquet_file_path)
+    
+    if not os.path.exists(filtered_output_parquet_path):
+        if not os.path.exists(os.path.dirname(output_parquet_path)):
+            os.makedirs(os.path.dirname(output_parquet_path))
```
Comment (jstaszel-agh) [inline filter_and_map_stations.py:85]: Using pathlib would make it easier.

**Review by jstaszel-agh — APPROVED** [review]
It is just the base for the analysis, but the code is good - I appreciate using parquet.
It is a pity that the group didn't start working on time.

File: download.py
Comment (barti2115) [reply download.py:1]: It is due to corruption of available zip files at site used for downloading data

File: read_data.py
Comment (barti2115) [reply read_data.py:1]: It is due to corruption of available zip files at site used for downloading data

**Pull request merged**